src/control/lib/control/pool.go

```diff
@@ -420,24 +421,71 @@ type (
 	}
 )
 
-func (pi *PoolInfo) MarshalJSON() ([]byte, error) {
-	if pi == nil {
+func (pqr *PoolQueryResp) MarshalJSON() ([]byte, error) {
+	if pqr == nil {
 		return []byte("null"), nil
 	}
 
-	rs, err := system.CreateRankSet(pi.RankSet)
-	if err != nil {
-		return nil, err
+	type Alias PoolQueryResp
+	aux := &struct {
+		EnabledRanks  *[]system.Rank `json:"enabled_ranks"`
+		DisabledRanks *[]system.Rank `json:"disabled_ranks"`
+		*Alias
+	}{
+		Alias: (*Alias)(pqr),
+	}
+
+	if pqr.EnabledRanks != nil {
+		ranks := pqr.EnabledRanks.Ranks()
+		aux.EnabledRanks = &ranks
+	}
+
+	if pqr.DisabledRanks != nil {
+		ranks := pqr.DisabledRanks.Ranks()
+		aux.DisabledRanks = &ranks
 	}
 
-	type Alias PoolInfo
-	return json.Marshal(&struct {
-		RankSet []system.Rank `json:"rank_set"`
+	return json.Marshal(&aux)
+}
+
+func unmarshallRankSet(ranks string) (*system.RankSet, error) {
```

Comment:
looking at the code it seems like this should be unnecessary, CreateRankSet() should handle cases "" and "[]", I presume you have tested this and it's not the case though?

Reply:
The case "[]" generated an error when I tested it.
For the case "", I have to confess that I did not tested it, but from what I understand from the code I will obtain an empty rank set which is not what I need in my specific case.